tests/utils/jobHandlerTestHelper.ts

```diff
@@ -56,20 +69,32 @@ export class JobHandlerTestHelper {
       this.repoConnector,
       this.logger,
       this.jobValidator,
-      this.repoBranchesRepo
+      this.repoBranchesRepo,
+      this.repoEntitlementsRepo
     );
     return this.jobHandler;
   }
 
-  setStageForDeploySuccess(isNextGen = true, prodDeploy = true, returnValue?: MockReturnValueOnce): string[] {
+  setStageForDeploySuccess(
+    isNextGen = true,
+    prodDeploy = true,
+    returnValue?: MockReturnValueOnce,
+    setupOptions: SetupOptions = {}
+  ): string[] {
     this.job.payload.repoBranches = TestDataProvider.getPublishBranchesContent(this.job);
     this.setupForSuccess(isNextGen);
     const publishOutput = TestDataProvider.getPublishOutputWithPurgedUrls(prodDeploy);
+
+    const { hasGatsbySiteId } = setupOptions;
+    if (hasGatsbySiteId) {
+      this.repoEntitlementsRepo.getGatsbySiteIdByGithubUsername.mockResolvedValue('gatsby_site_id');
+    }
+
     if (returnValue) {
-      this.jobCommandExecutor.execute.mockReturnValueOnce(returnValue);
+      this.jobCommandExecutor.execute.mockResolvedValue(returnValue);
     } else {
-      this.jobCommandExecutor.execute.mockReturnValueOnce({ status: 'success', output: 'Great work', error: null });
-      this.jobCommandExecutor.execute.mockReturnValueOnce({ status: 'Failed', output: publishOutput[0], error: null });
```

Comment:
Removing this line since it seems odd to overwrite the mocked value for success with a status representative of failure. Also `mockResolvedValue` is preferred for async functions.

Unfortunately, I could not do something similar to the function that sets up the job handler for failure, without needing to delve deeper into the code to refactor. Fixing this behavior for the other function is out of scope